dev/Deployment/DeploymentManager.cpp

```diff
@@ -378,17 +380,19 @@ namespace winrt::Microsoft::Windows::ApplicationModel::WindowsAppRuntime::implem
             }
 
             auto packageId{ AppModel::Identity::PackageIdentity::FromPackageFullName(packageFullName.c_str()) };
-            if (packageId.Version().Version >= targetVersion.Version)
+            if (packageId.Version().Version >= highestVersionFound)
             {
                 match = true;
-                if (packageId.Version().Version > targetVersion.Version)
-                {
-                    g_existingTargetPackagesIfHigherVersion.insert(std::make_pair(packageIdentifier, packageFullName));
-                }
-                break;
+                highestVersionFound = packageId.Version().Version;
+                highestVersionPackageFullName = packageFullName;
             }
         }
 
+        if (highestVersionFound > targetVersion.Version)
+        {
+            g_existingTargetPackagesIfHigherVersion[packageIdentifier] = highestVersionPackageFullName;
+        }
+
         RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), !match);
```

Comment:
You could replace the `match` variable with `highestVersionPackageFullName` is set, as the same meaning but no additional variable to needed to track "do we have a winner"

1. remove the `match` variable
2. change line 396 to `RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), highestVersionPackageFullName.empty());`